### ISSUE TITLE
E117: Unknown function: chansend

### ISSUE THREAD

**Issue by noonien** [fatih/vim-go#1999]
### What did you do? (required. The issue will be **closed** when not provided.)
Browsed go code.

### What did you expect to happen?
No errors.

### What happened instead?
In a quickfix window:
```
Error detected while processing function <SNR>48_auto_type_info[3]..go#tool#Info[3]..go#complete#Info[2]..<SNR>104_async_info[58]..go#job#Start:                                                                           
line   41:
E117: Unknown function: chansend
```

### Configuration (**MUST** fill this out):

* vim-go version:
master (5ec87ee2dd3443bb5d0c7f7331bec1d1ed60df9f)

* Vim version (first three lines from `:version`):
```
NVIM v0.2.2
Build type: RelWithDebInfo
LuaJIT 2.0.4
```
* Go version (`go version`):
`go version go1.11 linux/amd64`

@ https://github.com/fatih/vim-go/blob/5ec87ee2dd3443bb5d0c7f7331bec1d1ed60df9f/autoload/go/job.vim#L305
The go#Job#Start function calls `chansend` and `chanclose`, which i believe are depreacated functions, I believe it should be `jobsend` and `jobclose` instead.

**Comment by Carpetsmoker** [comment]
I think you got it reversed: `chansend()` is the new name, and `jobsend()` the old one; based on https://vi.stackexchange.com/a/16438/51

So try updating Neovim?

**Comment by noonien** [comment]
Reply from neovim dev team: https://github.com/neovim/neovim/issues/9051#issuecomment-424465302

I'm using the latest stable neovim version.

**Comment by jamessan** [comment]
0.3.1 is the latest stable version, so you're a couple versions behind.

**Comment by noonien** [comment]
Hmm, interesting, it doesn't seem to have made its way into the neovim ubuntu ppa. Sorry for not checking that out.

Either way, from what I can tell from the neovim dev response, one should check if those functions exists before calling them.

Thanks, have a great day!

**Comment by jamessan** [comment]
> We should let the user know that they need a never version of Nvim.

Why is 0.3.1 required?

**Comment by bfredl** [comment]
> exists('*chansend') ? 'chansend' : 'jobsend'

There no reason to do this, `jobsend()` continues to work on new versions. Just keep using `jobsend()` and `jobclose()` if the aim is so support older versions.

**Comment by bhcleek** [comment]
The aim is not to support older versions. Neovim is not making any guarantees about backward compatibility, and it's still pre-1.0.

**Comment by bhcleek** [comment]
Vim has `v:version`, but Neovim doesn't seem to update it correctly. Is there any easy way to get the version of Neovim besides parsing the output of `:version`?

**Comment by bfredl** [comment]
`v:version` still means vim version, otherwise would break a lot of plugins. `api_info().version` is neovim version.

**Comment by jamessan** [comment]
> Neovim is not making any guarantees about backward compatibility

This isn't true.  We've documented our compatibility requirements.  There shouldn't be compatibility breaks from release to release (although APIs are subject to change during development cycles).

As @bfredl pointed out, the legacy names jobsend/jobclose have been kept explicitly to _avoid_ breaking code.

**Comment by bhcleek** [comment]
How long will they be kept?

**Comment by jamessan** [comment]
We follow semantic versioning, so not until 1.x could we break an API.

**Comment by bhcleek** [comment]
Generally, pre-1.0 releases can break compatibility in semantic versioning. Can you point me to Neovim's documented api backward compatibility policy?

**Comment by jamessan** [comment]
https://neovim.io/doc/user/api.html#api-contract

**Comment by bhcleek** [comment]
Thanks @jamessan ! I'll see if we have anything in vim-go now or (or coming up, because I'm working on some additional functionality) that requires 0.3.1.

**Comment by bhcleek** [comment]
We've merged #2000 for now.
